robot/Cumulus/resources/GiftEntryPageObject.py

```diff
@@ -113,9 +97,200 @@ def store_template_record_id(self,template):
+                    if placeholder=="Select an Option":
+                        popup=npsp_lex_locators["flexipage-popup"]
+                        self.selenium.wait_until_page_contains_element(popup)
+                        option=npsp_lex_locators["span_button"].format(value)
+                        self.selenium.click_element(option)
+                    elif placeholder=="Search...":
+                        self.salesforce.populate_lookup_field(key,value)
+                    elif "Date" in field:
+                        locator=npsp_lex_locators["bge"]["datepicker_open"].format("Date")  
+                        self.selenium.wait_until_page_contains_element(locator)
+                        self.selenium.click_button(value)    
+                        self.selenium.wait_until_page_does_not_contain_element(locator,error="could not open datepicker")
+                    else:
+                        self.salesforce._populate_field(field_loc,value) 
+    
+    def add_field_bundle_to_new_section(self,bundle):
+        """Adds the specified field bundle to the template builder form if not already added"""
+        try:
+            self.selenium.click_button("Add Section")
+        except ElementClickInterceptedException:
+            self.selenium.execute_javascript("window.scrollBy(0,100)")
+            self.selenium.click_button("Add Section")
+        checkbox=npsp_lex_locators["gift_entry"]["field_input"].format(bundle,"input")
+        self.selenium.scroll_element_into_view(checkbox)
+        cb_loc=self.selenium.get_webelement(checkbox)
+        if not cb_loc.is_selected():
+            try:
+                self.salesforce._jsclick(checkbox) 
+            except ElementClickInterceptedException:
+                self.selenium.execute_javascript("window.scrollBy(0,100)")
+                self.salesforce._jsclick(checkbox)  
+
+    def add_batch_table_columns(self,*args):
+        """Adds specified batch columns to the visible section if they are not already added"""
+        mylist=[]
+        for i in args:
+            locator=npsp_lex_locators["gift_entry"]["duellist"].format("Available Fields",i)
+            if self.npsp.check_if_element_exists(locator):
+                mylist.append(i)
+        print(mylist)
+        if len(mylist)==0:
+            return
+        else:
+            for field in mylist:
+                locator = npsp_lex_locators["gift_entry"]["duellist"].format("Available Fields",field)
+                if mylist.index(field)==0:
+                    self.selenium.click_element(locator)
+                else:
+                    self.selenium.click_element(locator,'COMMAND')   
+            self.selenium.click_button("Move selection to Visible Fields")   
+            verify_field=npsp_lex_locators["gift_entry"]["duellist"].format("Available Fields",mylist[0])
+            self.selenium.wait_until_page_does_not_contain_element(verify_field)          
```

Comment:
How about simplifying the keyword to this ?? Test it out once if you think this works for you.

```
first_element = True
        for i in args:
            locator = npsp_lex_locators["gift_entry"]["duellist"].format("Available Fields", i)
            if self.npsp.check_if_element_exists(locator):
                if first_element:
                    self.selenium.click_element(locator)
                    first_element = False
                else:
                    self.selenium.click_element(locator, 'COMMAND')
```

Reply:
thanks safiya, didn't think about that. Added that along with slight additions for verification